### ISSUE TITLE
Verify python 3.7 still works and add it to github actions CI

### ISSUE THREAD

**Issue by jansel** [facebookresearch/torchdynamo#52]
It has been a while since I tested Python 3.7, so it is possible/likely something could be broken.

We should:
1) Verify it still works, and fix any issues in tests/torchbench
2) Add a github actions workflow [similar to the python 3.8 one](https://github.com/facebookresearch/torchdynamo/blob/main/.github/workflows/test-py38.yml) to prevent it from getting broken in the future.

**Comment by yanboliang** [comment]
I set up Python 3.7 env, run all unit tests and found the following 11 failed:
```
==================================================================================================== short test summary info ====================================================================================================
FAILED tests/test_dynamic_shapes.py::DynamicShapesMiscTests::test_no_grad_dynamic_shapes - torchdynamo.exc.InternalTorchDynamoError
FAILED tests/test_dynamic_shapes.py::DynamicShapesReproTests::test_get_parameter_dtype_dynamic_shapes - torchdynamo.exc.Unsupported: missing: SETUP_EXCEPT
FAILED tests/test_dynamic_shapes.py::DynamicShapesReproTests::test_maml_dynamic_shapes - AssertionError: 6 != 5
FAILED tests/test_dynamic_shapes.py::DynamicShapesReproTests::test_reformer_sorting_dynamic_shapes - torchdynamo.exc.InternalTorchDynamoError
FAILED tests/test_misc.py::MiscTests::test_no_grad - torchdynamo.exc.InternalTorchDynamoError
FAILED tests/test_repros.py::ReproTests::test_get_parameter_dtype - torchdynamo.exc.Unsupported: missing: SETUP_EXCEPT
FAILED tests/test_repros.py::ReproTests::test_maml - AssertionError: 5 != 4
FAILED tests/test_repros.py::ReproTests::test_reformer_sorting - torchdynamo.exc.InternalTorchDynamoError
FAILED tests/test_subgraphs.py::SubGraphTests::test_resume_with_no_grad1 - AssertionError: 1 != 2
FAILED tests/test_subgraphs.py::SubGraphTests::test_resume_with_no_grad2 - AssertionError: 1 != 3
FAILED tests/test_subgraphs.py::SubGraphTests::test_resume_with_no_grad3 - AssertionError: 1 != 2
========================================================================================== 11 failed, 429 passed, 9 skipped in 15.56s ===========================================================================================
```
These failures are caused by [this](https://docs.python.org/3/whatsnew/3.8.html#cpython-bytecode-changes) CPython bytecode change between 3.7 & 3.8. We have to:
* Replace ```BEGIN_FINALLY``` with any Python 3.7 instruction.
* As ```WITH_CLEANUP_START``` and ```END_FINALLY``` changed behavior between 3.7 & 3.8, we need to extend the implementation to support both version.

**Comment by jansel** [comment]
Thanks for looking into that @yanboliang.

The BEGIN_FINALLY/WITH_CLEANUP_START/END_FINALLY stuff is just trying to support:
```
with torch.no_grad():
   ...
```

Feel free to reach out to me if you get stuck.

**Comment by jansel** [comment]
Fixed by #143
